File: reservation_system/application/lib/models/group.dart
class Group {
  final String id;
  final String name;
  final String description;
  final int numberOfItems;
  final int numberOfReservations;
  final int overbooking;
  final String userId;

  Group(
    this.id,
    this.name,
    this.description,
    this.numberOfItems,
    this.numberOfReservations,
    this.overbooking,
    this.userId,
  );

  factory Group.fromJson(Map<String, dynamic> json) {
    return Group(
      json['groupId'],
      json['name'],
      json['description'],
      json['numberOfItems'],
      json['numberOfReservations'],
      int.parse(json['overbooking']),
      json['userId'],
    );
  }

  Map toJson() => {
        'groupId': id,
        'name': name,
        'description': description,
        'numberOfItems': numberOfItems,
        'numberOfReservations': numberOfReservations,
        'overbooking': overbooking,
        'userId': userId,
      };
}

File: chaincode/lib/group.js
'use strict';

class Group {
    /**
     *
     * Group
     *
     * Constructor for a Group object.
     *  
     * @param groupId - unique Id which corresponds to a created Group
     * @param name - Group name
     * @param description - Group description
     * @param overbooking - allowed overbooking
     * @param numberOfItems - number of Items in a Group
     * @param numberOfReservations - number of reserved Items in a Group
     * @param userId - id of a Group creator
     * @returns - Group object
     */
    constructor(groupId, name, description, overbooking, numberOfItems, numberOfReservations, userId) {

        this.groupId = groupId;
        this.name = name;
        this.description = description;
        this.overbooking = overbooking;
        this.numberOfItems = numberOfItems;
        this.numberOfReservations = numberOfReservations;
        this.userId = userId;
        this.type = 'group';
        if (this.__isContract) {
            delete this.__isContract;
        }
        return this;

    }

}
module.exports = Group;